agent_test.go

```diff
@@ -351,3 +366,67 @@ func TestConnectivityOnStartup(t *testing.T) {
 	<-aConnected
 	<-bConnected
 }
+
+func TestInboundValidity(t *testing.T) {
+	buildMsg := func(class stun.MessageClass, username, key string) *stun.Message {
+		msg, err := stun.Build(class, stun.MethodBinding, stun.GenerateTransactionID(),
+			&stun.Username{Username: username},
+			&stun.MessageIntegrity{
+				Key: []byte(key),
+			},
+			&stun.Fingerprint{},
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		return msg
+	}
+
+	remote := &net.UDPAddr{IP: net.ParseIP("172.17.0.3"), Port: 999}
+	local, err := NewCandidateHost("udp", net.ParseIP("192.168.0.2"), 777, 1)
+	if err != nil {
+		t.Fatalf("failed to create a new candidate: %v", err)
+	}
+
+	t.Run("Invalid Binding requests should be discarded", func(t *testing.T) {
+		a, err := NewAgent(&AgentConfig{})
+		if err != nil {
+			t.Fatalf("Error constructing ice.Agent")
+		}
+
+		a.handleInbound(buildMsg(stun.ClassRequest, "invalid", a.localPwd), local, remote)
+		if len(a.remoteCandidates) == 1 {
+			t.Fatal("Binding with invalid Username was able to create prflx candidate")
+		}
+
+		a.handleInbound(buildMsg(stun.ClassRequest, a.localUfrag+":"+a.remoteUfrag, "Invalid"), local, remote)
+		if len(a.remoteCandidates) == 1 {
+			t.Fatal("Binding with invalid MessageIntegrity was able to create prflx candidate")
+		}
+	})
+
+	t.Run("Invalid Binding requests should be discarded", func(t *testing.T) {
```

Comment:
is this description supposed to be the same as the last one?

Reply:
Fixed